pandas/core/sorting.py

```diff
@@ -218,7 +218,7 @@ def decons_group_index(comp_labels, shape):
     x = comp_labels
     for i in reversed(range(len(shape))):
         labels = (x - y) % (factor * shape[i]) // factor
-        np.putmask(labels, comp_labels < 0, -1)
+        np.putmask(labels, x < 0, -1)
```

Comment:
lets just do `labels[x < 0] = -1`, putmask has different semantics when the last arg is a listlike but thats not relevant here